rpcclient/infrastructure.go

```diff
@@ -1648,6 +1658,9 @@ func parseBitcoindVersion(version string) BackendVersion {
 	case version < bitcoind22Str:
 		return BitcoindPre22
 
+	case version < bitcoind24Str:
+		return BitcoindPre24
```

Comment:
IIUC, we need to start to check this in `btcwallet`? FWIW, didn't see it in the latest PR of yours in that repo.

Reply:
I think we'll need to check this in `lnd` at the callsite. If we check it in `btcwallet`, then it's another error to be returned to the caller. On the other hand, I think the `rpcclient` package should be put in `btcwallet` instead to keep `btcd` "pure" - came up with a draft PR for this when I was dealing with updating errors returned from `TestMempoolAccept`, will put more details there.

Reply:
>  I think the rpcclient package should be put in btcwallet instead to keep btcd "pure" - came up with a draft PR for this when I was dealing with updating errors returned from TestMempoolAccept, will put more details there.

Don't think I agree with this, `rpcclient` as is does generic JSON-RPC for conformant Bitcoin nodes. We even use the client to test directly against (in itests) the server-side implementation of the various RPC calls. 

I don't see why the check can't just be in the wallet though. We already have that check for `bitcoind` when we wan to use the newer set of RPC calls.